index.d.ts

```diff
@@ -202,6 +202,13 @@ declare namespace Conf {
 		*/
 		readonly accessPropertiesByDotNotation?: boolean;
 	}
+
+	interface ExtraType<T> {
+		name: string,
+		instance: T,
+		convertFrom: (val: T) => string | number,
```

Comment:
The return value here would be a good candidate for https://github.com/sindresorhus/type-fest/blob/eb110db80ccd74e8f5141cb52b3e99d3603399ce/source/basic.d.ts#L53